### PR TITLE
#17324: port Padding CPP Unit OP Tests to TTNN

### PR DESCRIPTION
**Pull request by llongTT**
### Ticket
Link to Github Issue

### Problem description
(https://github.com/tenstorrent/tt-metal/issues/17324)

### What's changed
Port Padding OP CPP Unit test to TTNN

### Checklist
- [x] Post commit CI passes(https://github.com/tenstorrent/tt-metal/actions/runs/13091704247)
- [ ] Blackhole Post commit (if applicable)
- [ ] Model regression CI testing passes (if applicable)
- [ ] Device performance regression CI testing passes (if applicable)
- [ ] **(For models and ops writers)** Full [new models](https://github.com/tenstorrent/tt-metal/actions/workflows/full-new-models-suite.yaml) tests passes
- [ ] New/Existing tests provide coverage for changes

### CODE REVIEW

File: tests/ttnn/unit_tests/test_pad_op.py [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+    output_tt = ttnn.pad(ttnn_input, padshape, [0, 0, 0, 0], value=0, use_multicore=use_multicore)
+    output_tt = ttnn.to_torch(output_tt)
+    assert output_tt.shape == torch.Size(padshape)
+
+    shape_diff = list(map(lambda x, y: x - y, padshape, shape))
+    print(shape_diff)
+    output_torch = torch.nn.functional.pad(torch_input, [0, shape_diff[-1], 0, shape_diff[-2]], value=0)
+    passing = torch.equal(output_tt, output_torch)
+    print("output_tt: ", output_tt)
+    print("output_torch: ", output_torch)
```
Comment (cfjchu) [inline tests/ttnn/unit_tests/test_pad_op.py:31]: remove these debug prints

**Review by cfjchu — APPROVED** [review]

**Review by bbradelTT — APPROVED** [review]

**Comment by blozano-tt** [comment]
@llongTT - all-post-commit has not passed on main since this PR was merged.

Does this test add significant runtime to ttnn python tests?

Here is the relevant run (I just launched the second retry). 

https://github.com/tenstorrent/tt-metal/actions/runs/13095068067

**Comment by llongTT** [comment]
> @llongTT - all-post-commit has not passed on main since this PR was merged. (Timeouts)
> 
> Does this test add significant runtime to ttnn python tests?
> 
> Here is the relevant run (I just launched the second retry).
> 
> https://github.com/tenstorrent/tt-metal/actions/runs/13095068067
> 
> If we can't get this passing we may have to revert.
> 
> FYI: @bkeith-TT @ttmchiou

It will definitely increase the run time as it's a new one, it's a rather small test and doesn't have much in it.
